### PR TITLE
[15.0] Migration of queue_job

### PR DESCRIPTION
**Pull request by guewen**
This is the work from @romi477 from https://github.com/OCA/queue/pull/378 rebased on 15.0

### CODE REVIEW

**Comment by ventor-dev** [comment]
@guewen so you will finish it or @romi477 should do smth? We need this module for our customer...

**Comment by pedrobaeza** [comment]
@guewen maybe you need to run the copier template (https://github.com/OCA/oca-addons-repo-template) over 15.0 for having the proper metafiles.

**Comment by guewen** [comment]
> maybe you need to run the copier template (https://github.com/OCA/oca-addons-repo-template) over 15.0 for having the proper metafiles.

Thanks @pedrobaeza! I fixed the base files now.

Reading the logs, it looks like it tries to install `queue_job`, which is a dependency of `test_queue_job`. The pip installable wheel for `queue_job` does not exist yet. @sbidoul did you see something like this already?

```
Run oca_install_addons
+ setuptools-odoo-get-requirements --include-addons --addons-dir .
+ cat test-requirements.txt
odoo-addon-queue_job>=15.0dev,<15.1dev
odoo>=15.0a,<15.1dev
requests
+ pip install -r test-requirements.txt
Looking in indexes: https://wheelhouse.odoo-community.org/oca-simple-and-pypi
ERROR: Could not find a version that satisfies the requirement odoo-addon-queue_job<15.1dev,>=15.0dev (from versions: none)
ERROR: No matching distribution found for odoo-addon-queue_job<15.1dev,>=15.0dev
Error: Process completed with exit code 1.
```

**Comment by sbidoul** [comment]
It's a bug. Addons of the current repo should not be added to test-requirements.txt. I'll look into it. (https://github.com/acsone/setuptools-odoo/issues/67)

**Comment by etobella** [comment]
Just one question. Can we delete test_queue_job and use `odoo_test_helper` in order to create the models in the tests. Something similar to https://github.com/OCA/edi/blob/14.0/edi_oca/tests/test_consumer_mixin.py ?

File: queue_job/fields.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class JobSerialized(fields.Field):
     def __init__(self, string=fields.Default, base_type=fields.Default, **kwargs):
         super().__init__(string=string, _base_type=base_type, **kwargs)
 
-    def _setup_attrs(self, model, name):
+    def _setup_attrs(self, model, name):  # pylint: disable=W8110
```
Comment (sbidoul) [inline queue_job/fields.py]: nit: the human-readable code is preferable

**Comment by sbidoul** [comment]
https://github.com/OCA/oca-ci/pull/16 should fix it. We can restart the tests here as soon as the oca-ci images are rebuilt.

**Comment by sbidoul** [comment]
:green_apple:

**Comment by guewen** [comment]
Wow, many thanks @sbidoul.

File: queue_job/fields.py (hunk 1 above)
Comment (guewen) [reply queue_job/fields.py]: Agree, I changed them

**Comment by guewen** [comment]
@etobella Why not, if anyone has time to dedicate to this...

**Review by simahawk — APPROVED** [review]
LG

**Comment by guewen** [comment]
> @guewen so you will finish it or @romi477 should do smth? We need this module for our customer...

@ventor-dev He can review and approve this PR :)

**Review by ventor-dev — APPROVED** [review]

**Review by romi477 — APPROVED** [review]

**Comment by ventor-dev** [comment]
@guewen @sbidoul when do you think that can be merged?

Also will it be also published to Odoo Apps?

**Pull request merged**